source/userguide/kb.rst

```diff
@@ -813,3 +813,122 @@ We can perform the same search using the :meth:`query()` API as well.
 	To delete and reindex your data, follow the steps mentioned :ref:`here <import_kb>`.
 
 
+.. _semantic_embeddings:
+
+Leveraging semantic embeddings
+------------------------------
+Up until now, the described Question Answering relies on text based retrieval. Deep learning based dense embeddings (character, word, or sentence) are in many cases better at capturing semantic information than traditional sparse vectors. Pretrained or fine-tuned embeddings can be used to find the best match in the Knowledge Base even if the search token wasn’t present in the uploaded data.
+
+To leverage semantic embeddings in search, the first step is to generate the embeddings for your desired Knowledge Base fields. You can decide to use one of the provided embedders, or you can use your own. If you app consists of mainly standard English vocabulary, one of provided embedders may work well enough, but if the text you are searching against has quite a bit of domain specific vocabulary, you may benefit from training or fine tuning your own embedder on your data.
+
+To use one of the built in embedders, you can simply specify an emedder ``model_type`` in your Question Answering config file, specify which model you would like to use ``model_settings``, and specify which fields you would like to generate embeddings for in ``embedding_fields``. The embedding fields parameter takes a dictionary where the key is the name of your index, and the value is a list of field names or regexes to match against the field names for that index. Using the HR Assistant blueprint as an example, here is what the config would look like.
+
+.. code:: python
+
+  QUESTION_ANSWERER_CONFIG = {
+      "model_type": "embedder",
+      "model_settings": {
+          "embedder_type": "bert",
+          "embedding_fields": {
+              "faq_data": ["question", "answer"]
+          }
+      }
+  }
+
+There are three available model types which leverage semantic embedders. The specified ``model_type`` is also the default ``query_type`` for all question answering calls for your application, but you can always pass in a different query type to your ``qa.get()`` command as desired.
+
+  +-----------------------+------------------------------------------------------------------------------------------------------------+
+  | Model type            | Description                                                                                                |
```

Comment:
Could you change the descriptions to:

Only leverage deep semantic embedders. This option allows for using deep semantic embedders like Sentence-BERT or GloVe for doing vector-based retrieval.

Leverage a combination of deep semantic embedders and text signals in a way that's optimized for search on keywords or short spans of text. GloVe is preferable for this use case.

Leverage a combination of deep semantic embedders and text signals in a way that's optimized for search on larger paragraphs or passages of unstructured text. Sentence-BERT is preferable for this use case.

Reply:
Fixed